signac/synced_collections/backends/collection_mongodb.py

```diff
@@ -0,0 +1,187 @@
+
+    Parameters
+    ----------
+    collection : :py:class:`pymongo.collection.Collection`
+        The MongoDB client in which to store data.
+    uid : dict
+        The unique key-value mapping added to the data and stored in the document
+        so that it is uniquely identifiable in the MongoDB collection.
+
+    Warnings
+    --------
+    The user is responsible for providing a unique id such that there are no
+    possible collisions between different :class:`SyncedCollection` instances
+    stored in the same MongoDB collection. Failure to do so may result data
+    corruption if multiple documents are found to be apparently associated with
+    a given :class:`SyncedCollection`.
+
+    """
+
+    _backend = __name__  # type: ignore
+
+    def __init__(self, collection=None, uid=None, parent=None, **kwargs):
+        if not MONGO:
+            raise RuntimeError(
+                "The PyMongo package must be installed to use the MongoDBCollection."
+            )
+
+        self._collection = collection
+        self._uid = uid
+        super().__init__(parent=parent, **kwargs)
+
+    def _load_from_resource(self):
+        """Load the data from a MongoDB document.
+
+        Returns
+        -------
+        Collection or None
+            An equivalent unsynced collection satisfying :meth:`is_base_type` that
+            contains the data in the MongoDB database. Will return None if no data
+            was found in the database.
+
+        """
+        blob = self._collection.find_one(self._uid)
+        return blob["data"] if blob is not None else None
+
+    def _save_to_resource(self):
+        """Write the data to a MongoDB document."""
+        data = self._to_base()
+        data_to_insert = {**self._uid, "data": data}
+        try:
+            self._collection.replace_one(self._uid, data_to_insert, True)
+        except bson.errors.InvalidDocument as err:
+            raise TypeError(str(err))
+
+    @property
+    def collection(self):
+        """pymongo.collection.Collection: Get the collection being synced to."""
+        return self._collection
+
+    @property
+    def uid(self):  # noqa: D401
+        """dict: Get the unique mapping used to identify this collection."""
+        return self._uid
+
+    def __deepcopy__(self, memo):
+        # The underlying MongoDB collection cannot be deepcopied.
+        raise TypeError("MongoDBCollection does not support deepcopying.")
+
+
+class MongoDBDict(MongoDBCollection, SyncedAttrDict):
+    """A dict-like mapping interface to a persistent document in a MongoDB collection.
+
+    .. code-block:: python
+
+        doc = MongoDBDict('data')
+        doc['foo'] = "bar"
+        assert doc.foo == doc['foo'] == "bar"
+        assert 'foo' in doc
+        del doc['foo']
+
+    .. code-block:: python
+
+        >>> doc['foo'] = dict(bar=True)
+        >>> doc
+        {'foo': {'bar': True}}
+        >>> doc.foo.bar = False
+        {'foo': {'bar': False}}
```

Comment:
Fix these code blocks (match whether to use doctest style / Examples heading) or remove them. Also applies to all subsequent example code blocks in all collection classes.